events-api-ht-3/utils.js

```diff
@@ -0,0 +1,114 @@
+const fs = require('fs');
+const Transform = require("stream").Transform;
+
+const csvFile = "events.csv";
+const lineBreak = '\n';
+
+
+async function getEventsByLocation(location) {
+  const fileContent = await readFile();
+
+  return location ? fileContent.filter(event => event.location.toLowerCase() === location.toLowerCase()) : fileContent;
+}
+
+async function getEventById(id) {
+  const fileContent = await readFile();
+
+  return fileContent.filter(event => event.id === id)[0];
+}
+
+function addNewEvent(reqBody) {
+  let logStream = fs.createWriteStream(csvFile, {flags: 'a'});
+
+  const id = new Date().getTime();
+
+  const newEvent = Object.values(reqBody);
```

Comment:
with such approach we are dependent on fields order. If smbd pass data in different order we may write data to incorrect fields:

```
Object.values({
    "title": "City day",
    "location": "Rivne",
    "date": "2021-06-04",
    "guests": "500"
}) => ["City day", "Rivne", "2021-06-04", "500"]

Object.values({
    "location": "Rivne",
    "title": "City day",
    "date": "2021-06-04",
    "guests": "500"
}) => ["Rivne", "City day", "2021-06-04", "500"]
```